server/libs/infra/src/infra.config.ts

```diff
@@ -3,17 +3,34 @@ import { BullModuleOptions } from '@nestjs/bull';
 import { RedisOptions } from 'ioredis';
 import { ConfigurationOptions } from 'typesense/lib/Typesense/Configuration';
 
-export const redisConfig: RedisOptions = {
-  host: process.env.REDIS_HOSTNAME || 'immich_redis',
-  port: parseInt(process.env.REDIS_PORT || '6379'),
-  db: parseInt(process.env.REDIS_DBINDEX || '0'),
-  password: process.env.REDIS_PASSWORD || undefined,
-  path: process.env.REDIS_SOCKET || undefined,
-};
+export function redisConfig(): RedisOptions {
```

Comment:
I'm fine with a function that returns the config, but can you still just export the config here? So like
```typescript
function parseRedisConfig(): RedisOptions {
...
export const redisConfig: RedisOptions = parseRedisConfig();
```